Builds failing using musl libc

Recently started getting these errors when building:

```bash
$ make -C src/
make: Entering directory 'src'
     CC queue.ol
In file included from queue.c:14:
include/liburing.h:162:39: error: unknown type name ‘cpu_set_t’
  162 |                                 const cpu_set_t *mask);
      |                                       ^~~~~~~~~
make: *** [Makefile:45: queue.ol] Error 1
make: Leaving directory 'liburing/src'
```

It seems that `-D_GNU_SOURCE` isn't defined, and musl libc requires this before defining `cpu_set_t`.

This patch fixes it, but not 100% sure if this is the right way and if it covers cpp or not:

```diff
diff --git a/src/Makefile b/src/Makefile
index dfca826..0614f30 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -6,7 +6,7 @@ libdevdir ?= $(prefix)/lib
 CPPFLAGS ?=
 override CPPFLAGS += -Iinclude/ -include ../config-host.h
 CFLAGS ?= -g -fomit-frame-pointer -O2
-override CFLAGS += -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare
+override CFLAGS += -D_GNU_SOURCE -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare
 SO_CFLAGS=-fPIC $(CFLAGS)
 L_CFLAGS=$(CFLAGS)
 LINK_FLAGS=
```

--- a/examples/Makefile
+++ b/examples/Makefile
@@ -2,7 +2,7 @@ CPPFLAGS ?=
 override CPPFLAGS += -D_GNU_SOURCE -I../src/include/
 CFLAGS ?= -g -O2
 XCFLAGS =
-override CFLAGS += -Wall -L../src/
+override CFLAGS += -D_GNU_SOURCE -Wall -L../src/
 
 include ../Makefile.quiet
 

--- a/src/Makefile
+++ b/src/Makefile
@@ -4,9 +4,11 @@ libdir ?= $(prefix)/lib
 libdevdir ?= $(prefix)/lib
 
 CPPFLAGS ?=
-override CPPFLAGS += -Iinclude/ -include ../config-host.h
+override CPPFLAGS += -D_GNU_SOURCE \
+	-Iinclude/ -include ../config-host.h
 CFLAGS ?= -g -fomit-frame-pointer -O2
-override CFLAGS += -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare
+override CFLAGS += -D_GNU_SOURCE \
+	-Wall -Wextra -Wno-unused-parameter -Wno-sign-compare
 SO_CFLAGS=-fPIC $(CFLAGS)
 L_CFLAGS=$(CFLAGS)
 LINK_FLAGS=

--- a/test/Makefile
+++ b/test/Makefile
@@ -12,7 +12,8 @@ override CPPFLAGS += -D_GNU_SOURCE -D__SANE_USERSPACE_TYPES__ \
 	-I../src/include/ -include ../config-host.h
 CFLAGS ?= -g -O2
 XCFLAGS =
-override CFLAGS += -Wall -Wextra -Wno-unused-parameter -Wno-sign-compare \
+override CFLAGS += -D_GNU_SOURCE \
+	-Wall -Wextra -Wno-unused-parameter -Wno-sign-compare \
 	-L../src/
 
 ifdef CONFIG_HAVE_STRINGOP_OVERFLOW

